Sources/ArcGISToolkit/FloorFilter/SiteFacilityPromptViewController.swift

```diff
@@ -0,0 +1,272 @@
+    
+    private func getFilteredFacilities() -> [AGSFloorFacility] {
+        return isSearchActive ? filteredSearchFacilities : viewModel.facilities
+    }
+    
+    private func getFilteredSites() -> [AGSFloorSite] {
+        return isSearchActive ? filteredSearchSites : viewModel.sites
+    }
+    
+    override public func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        isShowingFacilities = (viewModel.sites.isEmpty || viewModel.selectedFacility != nil) ? true : false
+        initializeButtonsClickListeners()
+        updatePromptViewHeight()
+        updatePromptTitle()
+    }
+    
+    override public func viewDidLoad() {
+        super.viewDidLoad()
+        initializeSiteFacilityTableView()
+        initializeSiteFacilitySearchBar()
+        initializeButtonsClickListeners()
+       
+        updatePromptViewHeight()
+        updatePromptTitle()
+    }
+    
+    private func initializeButtonsClickListeners() {
+        closeBtn.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(self.closeSiteFacilityPrompt)))
+        closeBtn.isUserInteractionEnabled = true
+        backBtn.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(self.backButtonPressed)))
+        backBtn.isUserInteractionEnabled = true
+    }
+    
+    private func updatePromptViewHeight() {
+        self.designableViewHeight.constant = 380
+        bottomConstraint.constant = ((UIScreen.main.bounds.height / 2) - (self.designableViewHeight.constant / 2))
+        originalYPositionForPanel = bottomConstraint.constant
+    }
+    
+    @objc func closeSiteFacilityPrompt() {
+        self.dismiss(animated: true, completion: nil)
+    }
+    
+    @objc func backButtonPressed() {
+        isShowingFacilities = false
+        siteFacilityTableView?.reloadData()
+    }
+    
+    private func updatePromptTitle() {
+        promptTitle.font = UIFont.boldSystemFont(ofSize: 17)
+        promptSubtitle.isHidden = true
+        promptSubtitle.text = ""
+        if (isShowingFacilities) {
+            // Add the subtitle when showing facilities
+            promptSubtitle.isHidden = false
+            promptSubtitle.text = "Select a Facility"
+            promptTitle.text = "\(viewModel.selectedSite?.name ?? "")"
+            backBtn.isHidden = false
+        } else {
+            promptTitle?.text = "\(viewModel.selectedSite?.name ?? "Select a Site")"
+            backBtn?.isHidden = true
+        }
+    }
+}
+
+/// Extension for Search Bar functions
+extension SiteFacilityPromptViewController: UISearchBarDelegate {
+    private func initializeSiteFacilitySearchBar() {
+        siteFacilitySearchBar.delegate = self
+        resetSearchFilteredResults()
+        UIBarButtonItem.appearance(whenContainedInInstancesOf: [UISearchBar.self]).setTitleTextAttributes([NSAttributedString.Key(rawValue: NSAttributedString.Key.foregroundColor.rawValue): UIColor.customBlue], for: .normal)
```

Comment:
Please do not customize the appearance of `UIBarButtonItem` like this as it will affect all bar button items in all search bars from this point forward--not just in the toolkit, but in the user's app as well!

FYI: Taking an instance of a raw representable type, getting its raw value, and then using that to construct an instance of that same type is unnecessary. Just use the instance directly: `setTitleTextAttributes([.foregroundColor: UIColor.customBlue], for: .normal)`.